src/app/services/species.service.spec.ts

```diff
@@ -23,11 +22,9 @@ describe('SpeciesService', () => {
   });
 
   it('should get a species by id', () => {
-    httpClientSpy.get.and.returnValue(of(species[0]));
-    service.getById(1).subscribe(data => {
-      expect(data.name).toContain('Human');
-      expect(data.classification).toContain('mammal');
-      expect(data.designation).toContain('sentient');
+    httpClientSpy.get.and.returnValue(of(species.results[0]));
+    service.getById(0).subscribe(data => {
```

Comment:
This is very unusual to use 0 as an ID. I expected to see `species.results[0].id` equivalent here so that when you read the test its clear whats going on.

Make your test fakes are a reasonable reflection of real world data. Its a little subjective, but the key is to have tests that make sense when you read them.

Reply:
Are you referring to how we access the data in the json file on line 25?

Reply:
To L26 highlighted above where you' have `getById(0)`.

Reply:
We troubleshooted and resolved the issue. Ran the test several times with no error at all.